libqtile/widget/widgetbox.py

```diff
@@ -0,0 +1,189 @@
+    """A widget to declutter your bar.
+
+    WidgetBox is a widget that hides widgets by default but shows them when
+    the box is opened.
+
+    Widgets that are hidden will still update etc. as if they were on the main
+    bar.
+
+    Button clicks are passed to widgets when they are visible so callbacks will
+    work.
+
+    Widgets in the box alsos remain accessible via command interfaces.
+
+    Widgets can only be added to the box via the configuration file. The widget
+    is conigured by adding widgets to the "widgets" parameter as follows:
+
+        widget.WidgetBox(widgets=[
+            widget.TextBox(text="This widget is in the box"),
+            widget.Memory(),
+            ]),
+    """
+    orientations = base.ORIENTATION_HORIZONTAL
+    defaults = [
+        (
+            "font",
+            "sans",
+            "Text font"
+        ),
+        (
+            "fontsize",
+            None,
+            "Font pixel size. Calculated if None."
+        ),
+        (
+            "fontshadow",
+            None,
+            "font shadow color, default is None(no shadow)"
+        ),
+        (
+            "foreground",
+            "#ffffff",
+            "Foreground colour."
+        ),
+        (
+            "close_button_location",
+            "left",
+            "Location of close button when box open ('left' or 'right')"
+        ),
+        (
+            "text_closed",
+            "[<]",
+            "Text when box is closed"
+        ),
+        (
+            "text_open",
+            "[>]",
+            "Text when box is open"
+        ),
+    ]
+
+    def __init__(self, widgets=list(), **config):
+        base._Widget.__init__(self, bar.CALCULATED, **config)
+        self.add_defaults(WidgetBox.defaults)
+        self.box_is_open = False
+        self._widgets = widgets
+        self.add_callbacks({"Button1": self.cmd_toggle})
+
+        if self.close_button_location not in ["left", "right"]:
+            val = self.close_button_location
+            msg = "Invalid value for 'close_button_location': {}".format(val)
+            logger.warning(msg)
+            self.close_button_location = "left"
+
+    def _configure(self, qtile, bar):
+        base._Widget._configure(self, qtile, bar)
+
+        self.layout = self.drawer.textlayout(
+            self.text_closed,
+            self.foreground,
+            self.font,
+            self.fontsize,
+            self.fontshadow,
+            markup=False,
+        )
+
+        for idx, w in enumerate(self._widgets):
+            if w.configured:
+                w = w.create_mirror()
+                self._widgets[idx] = w
+            self.qtile.register_widget(w)
+            w._configure(self.qtile, self.bar)
+
+            # In case the widget is mirrored, we need to draw it once so the
+            # mirror can copy the surface but draw it off screen
+            w.offsetx = self.bar.width
+            self.qtile.call_soon(w.draw)
```

Comment:
Does this continue to allow mirror widgets that mirror a widget that is hidden inside the widgetbox to update when the hidden one tries to update?

Reply:
It should do, yes.

The initial `draw` is called here because, without it, the mirror widget is blank until the box is opened for the first time. Calling `draw` means that the widget's drawer surface is drawn so the mirror has something to copy.

If we use the clock widget as an example and put the original in the box and the widget outside it, the mirror continues to show the correct time even if you don't open the box. Is that the issue you're concerned about or have I missed a different scenario?